### DISCUSSION TITLE
How about adding more CSS-classes for custom theme?

### DISCUSSION THREAD

**Discussion by fulgor** [datenstrom/community#302] [Ask a question] [answered]
Yellow's Sitemap-Plugin is a must-have. However, with more and more pages, the classic `ul`-list gets very long. Thus, I changed `<ul>` to `<ul class="sitemap">` in [content-sitemap.php](
https://github.com/datenstrom/yellow-plugins/blob/master/sitemap/content-sitemap.php). Then I added the following lines to [flatsite.css](https://github.com/datenstrom/yellow/blob/master/system/themes/assets/flatsite.css):
```
ul.sitemap li  { display: inline; list-style-type: none; }
ul.sitemap li:after {  content: " \2014 "; }
ul.sitemap li:last-child:after {  content: ""; }
```
Now, all `<li>`-items are inline, separated by an mdash ( hex "\2014").

I do **not** propose to change flatsite.css in yellow-core. This belongs to theming. However, I propose to add classes in plugins, so updates would not mean to rewrite this kind of changes.

**Comment by BsNoSi** [comment]
I think, an inline "list" is confusing.

Maybe something like this would be better. I have to admit, that this requires modification of the module, because a sliglty modified html output is needed. 

```
<html>
<head>
 <style type="text/css">
    .sitemap ul li {display:none;}
    .sitemap ul em {font-size:1.5em;font-weight:bold; color:red;}
    .sitemap ul:focus li {display:block;} 
  </style>
</head>
<body>
<div class="sitemap">
<ul tabindex="0"><em>A</em>
  <li>Ah</li>
  <li>Ah!</li>

</ul>
<ul tabindex="0"><em>B</em>
  <li>Bla</li>
  <li>Bla!</li>
</ul>
<ul tabindex="0"><em>C</em>
  <li>Cool.</li>
  <li>Cool!</li>
</ul>
</div>
</div>
</body>
</html>
```

The result of this (working)  sample is a list, that hides all `li` entries exept the enclosing `ul tabindex="0"` is focused. This works with `TAB` key, mouse click an should work on a tablet/smart phone, too (not tested).

**Comment by BsNoSi** [comment]
The previous version will not work propperly, if you *really* use tab, because jumping to entries will close the list. To improve this, you may use this (working) sample. Some more html code, but stable.

```
<html>
<head>
 <style type="text/css">
    .sitemap label {display:block;font-size:1.5em;color:green;font-weight:bold;}
    .sitemap .checkbox {display:none;}
    .sitemap .checkbox:checked + ul {display:block;}
    .sitemap ul {display:none;}
    .sitemap ul {font-size:0.8em;color:red; font-weight:normal;list-style:none;} 
  </style>
</head>
<body>
<div class="sitemap">
<label><input type="radio" name="1" class="checkbox" />A
<ul> 
  <li><a href="">Ah</a></li>
  <li><a href="">Ah</a></li>
</ul>
</label>
<label><input type="radio" name="1" class="checkbox" />B
<ul>
  <li><a href="">Ah</a></li>
  <li><a href="">Ah</a></li>
</ul>
</label>
<label><input type="radio" name="1" class="checkbox" />C
<ul>
  <li>Cool.</li>
  <li>Cool!</li>
</ul>
</label>
</div>
</body>
</html>
```
This solution depends on a (hidden) checkbox status (and requires modification of the module, too).

**Comment by markseuffert** [comment, marked as answer]
Hi Fulgor,

it's a good idea. If more developers want this we can and add more CSS-classes. I looked trough the existing code. At the moment there are not many CSS-classes in plugins, at least not in the deeper elements of the HTML code. There should already be useful information in the higher elements of the HTML code that you can use for customising themes, especially the CSS-class of the template.

Here's the HTML code that's generated by the [default sitemap](https://github.com/annaesvensson/yellow-sitemap):

``` php
<!DOCTYPE html>
<html>
  <head>…</head>
  <body>
    <div class="page template-sitemap">
      <div class="header">…</div>
      <div class="content">…</div>        <------ ul is inside content
      <div class="footer">…</div>
    </div>
  </body>
</html>
```

Here's an CSS example that you could use in your own theme:

```
.template-sitemap .content ul li  { display: inline; list-style-type: none; }
.template-sitemap .content ul li:after { content: " \2014 "; }
.template-sitemap .content ul li:last-child:after { content: ""; }
```

What do you think? Suggestions are very welcome.

**Comment by fulgor** [comment]
Hello Markseu,
I should have taken one step back, to see the whole picture.
Your way is more elegant. Thank you!

**Comment by markseuffert** [comment]
Thank you for the good idea. Let us know if you have more.
